# Importación y exploración

In [26]:
import pandas as pd

In [27]:
movies = pd.read_csv('../data/raw/0_inicial/movies.csv')
movies.drop('Unnamed: 0', inplace=True, axis=1)
movies.head()

Release Year                    Title Origin/Ethnicity          Director  \
0          1903      alice in wonderland         American    Cecil Hepworth   
1          1903  the great train robbery         American   Edwin S. Porter   
2          1910             frankenstein         American  J. Searle Dawley   
3          1910               pocahontas         American           Unknown   
4          1912  dr. jekyll and mr. hyde         American  Lucius Henderson   

                                                Cast          Genre  \
0                                          May Clark        unknown   
1                                                NaN        western   
2  Augustus Phillips, Charles Stanton Ogle, Mary ...        unknown   
3       Anna Rosemond, George Barnes, Frank H. Crane  short fantasy   
4                                        James Cruze         horror   

                                           Wiki Page  \
0  https://en.wikipedia.org/wiki/Alice_in_Wonderl...   
1  https://en.wikipedia.org/wiki/The_Great_Train_...   
2  https://en.wikipedia.org/wiki/Frankenstein_(19...   
3  https://en.wikipedia.org/wiki/Pocahontas_(1910...   
4  https://en.wikipedia.org/wiki/Dr._Jekyll_and_M...   

                                                Plot  movieId  \
0  Alice follows a large white rabbit down a "Rab...   1032.0   
1  The film opens with two bandits breaking into ...   6990.0   
2  Described as "a liberal adaptation of Mrs. She...   2648.0   
3  Though the film is presumed lost, a synopsis s...     48.0   
4  White-haired Dr. Jekyll has secretly locked hi...   7208.0   

                     title                                        genres  
0      alice in wonderland  Adventure|Animation|Children|Fantasy|Musical  
1  the great train robbery           Action|Adventure|Comedy|Crime|Drama  
2             frankenstein                           Drama|Horror|Sci-Fi  
3               pocahontas      Animation|Children|Drama|Musical|Romance  
4  dr. jekyll and mr. hyde                                  Drama|Horror

In [28]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4750 entries, 0 to 4749
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Release Year      4750 non-null   int64  
 1   Title             4750 non-null   object 
 2   Origin/Ethnicity  4750 non-null   object 
 3   Director          4750 non-null   object 
 4   Cast              4697 non-null   object 
 5   Genre             4750 non-null   object 
 6   Wiki Page         4750 non-null   object 
 7   Plot              4750 non-null   object 
 8   movieId           4750 non-null   float64
 9   title             4750 non-null   object 
 10  genres            4750 non-null   object 
dtypes: float64(1), int64(1), object(9)
memory usage: 408.3+ KB


# Analizando los géneros en la columna 'genres'

In [32]:
print(movies['genres'].head())

print(len(movies['genres'].unique()))

0    Adventure|Animation|Children|Fantasy|Musical
1             Action|Adventure|Comedy|Crime|Drama
2                             Drama|Horror|Sci-Fi
3        Animation|Children|Drama|Musical|Romance
4                                    Drama|Horror
Name: genres, dtype: object
639


Hay 639 entradas distintas de géneros, pero en realidad son combinaciones de géneros.
¿Cuántos géneros únicos hay en realidad?

In [61]:
# Voy a generar una lista con los géneros únicos.
generos_separados = [generos.split(sep='|') for generos in movies['genres']]

generos_simples = [genero for lista in generos_separados for genero in lista]
generos_unicos = list(set(generos_simples))

print(generos_unicos)
print(len(generos_unicos))

['Fantasy', 'Western', 'Horror', 'War', 'Film-Noir', 'Adventure', 'Thriller', 'Drama', 'IMAX', '(no genres listed)', 'Sci-Fi', 'Crime', 'Mystery', 'Comedy', 'Musical', 'Children', 'Action', 'Romance', 'Documentary', 'Animation']
20


Hay 20 géneros únicos. Una cantidad mucho más lógica y manejable para poder usar como metadata de las películas.

# Conclusión
Teniendo en mente que el objetivo inicial es hacer un recomendador de películas en base a los plots, pero que a futuro se pueda mejorar agregando metadata de las películas, se recomienda:
* Eliminar las columnas 'Genre', 'Wiki Page' y 'title', ya que no aportan metadata importante o repiten información de otras columnas,
* Cuando se quiera utilizar el género de la pelíucla como metadata, usar la lista de generos únicos